test/singleton.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+test('singleton', function() {
+  eq(S.singleton('foo', 42), {foo: 42});
+});
```

Comment:
Nits:

  - Use two empty lines to separate imports from the rest of the module.
  - Include one empty line at the beginning of the anonymous function's body, and one at the end.

In other words:

```javascript
'use strict';

... imports ...


test('...', function() {

  ... assertions ...

});
```

I'm sorry for nitpicking. I value internal consistency. :)